### PR TITLE
Deploy metallb through metallb-operator

### PR DESCRIPTION
**Pull request by sabinaaledort**

### CODE REVIEW

**Comment by sabinaaledort** [comment]
waiting for https://github.com/openshift/metallb-operator/pull/37/commits/a2d7303d9ad51e5939a3b8cae0015dbdcb36b0bf

**Comment by sabinaaledort** [comment]
/retest
failed to create the OCP cluster (unrelated to my change)

**Comment by sabinaaledort** [comment]
/retest
failed to create the OCP cluster (unrelated to my change)

File: metallb/deploy_operator.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/bash
+
+METALLB_OPERATOR_REPO=${METALLB_OPERATOR_REPO:-"https://github.com/metallb/metallb-operator.git"}
```
Comment (oribon) [inline metallb/deploy_operator.sh:3]: shouldn't we point to the openshift repo?

Comment (sabinaaledort) [reply metallb/deploy_operator.sh:3]: i wanted to stick to the e2e tests default repo (the u/s one) here:
https://github.com/openshift-metal3/dev-scripts/blob/eeb48a787cc2fca426babdb2eb5e80779e9ce945/metallb/run_e2e.sh#L3
so i thought to set the repo to the openshift one in the CI config here:
https://github.com/openshift/release/blob/928719efeaa06d43c59046bd0aca434a283bfd18/ci-operator/config/openshift/metallb/openshift-metallb-main.yaml#L60

File: 06_create_cluster.sh [hunk 2]
```diff
@@ -51,22 +51,15 @@ fi
...
+	pushd metallb
+	./configure_metallb.sh
+	popd
 
-	if [[ ${ENABLE_METALLB_MODE} == "l2" ]]; then
-		pushd metallb
-		./metallb_l2.sh
-		popd
-	elif [[ ${ENABLE_METALLB_MODE} == "bgp" ]]; then
+	if [[ ${ENABLE_METALLB_MODE} == "bgp" ]]; then
```
Comment (fedepaol) [inline 06_create_cluster.sh:58]: Do we need metallb modes anymore?

File: metallb/deploy_operator.sh (hunk 1 above)
Comment (fedepaol) [reply metallb/deploy_operator.sh:3]: I'd explicitly pin to a specific commit (either here or under release)

File: metallb/deploy_operator.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+cd metallb-operator
+
+podman build . -f ${METALLB_OPERATOR_DOCKER_FILE} -t ${OPERATOR_IMAGE} --authfile ${PERSONAL_PULL_SECRET}
+podman push ${OPERATOR_IMAGE} --tls-verify=false --authfile ${PULL_SECRET_FILE}
+
+# install yq v4 for metallb deployment
+yq_tmp_dir=$(mktemp -d)
+cd $yq_tmp_dir
+go mod init tmp
+GO111MODULE=on go get github.com/mikefarah/yq/v4
```
Comment (fedepaol) [inline metallb/deploy_operator.sh:24]: you can use go install here (same as u/s)

File: metallb/deploy_operator.sh [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+rm -rf $yq_tmp_dir
+cd -
+
+PATH="$GOPATH:$PATH" ENABLE_OPERATOR_WEBHOOK=true KUSTOMIZE_DEPLOY_DIR="config/webhook-on-openshift" IMG=${OPERATOR_IMAGE} make deploy
```
Comment (fedepaol) [inline metallb/deploy_operator.sh:28]: do we have a way to override metallb images too? I think we'd need the latest CI gives us

File: 06_create_cluster.sh (hunk 2 above)
Comment (sabinaaledort) [reply 06_create_cluster.sh:58]: i wanted to leave in this commit the start frr script as it's unrelated to the deployment method change. maybe i better remove the modes in my second PR
https://github.com/openshift-metal3/dev-scripts/pull/1305

File: metallb/deploy_operator.sh (hunk 4 above)
Comment (sabinaaledort) [reply metallb/deploy_operator.sh:28]: i think we will have to add a script that's setting the images like we do in the bump versions script
https://github.com/metallb/metallb-operator/blob/4d52b863b2a4764d37bf37acacf49eb2a74eee79/hack/bump_versions.sh#L6

**Comment by fedepaol** [comment]
cc @russellb

File: 06_create_cluster.sh [hunk 5]
```diff
@@ -57,16 +57,15 @@ if [[ ! -z "${ENABLE_METALLB_MODE}" ]]; then
...
 	fi
 
-	if [[ ${ENABLE_METALLB_MODE} == "l2" ]]; then
-		pushd metallb
-		./metallb_l2.sh
-		popd
-	elif [[ ${ENABLE_METALLB_MODE} == "bgp" ]]; then
+	pushd metallb
+	./configure_metallb.sh
+	popd
```
Comment (russellb) [inline 06_create_cluster.sh:62]: I don't think metallb should be deployed unconditionally.  That will affect a bunch of CI jobs that we shouldn't mess with if we're not explicitly testing metallb.

**Review by russellb — CHANGES_REQUESTED** [review]

File: 06_create_cluster.sh (hunk 5 above)
Comment (russellb) [reply 06_create_cluster.sh:62]: sorry, i missed that this was inside a conditional block on ENABLE_METALLB_MODE ... i blame the code being collapsed on github :-D

**Comment by russellb** [comment]
should be harmless to everyone except those turning on metallb

/approve
/lgtm

**Pull request merged**